### PR TITLE
Ease other framework applications generation

### PR DESCRIPTION
**Pull request by sdelamo**
The goal of this PR is to support easier application generation of [other framework applications in Micronaut Guides](https://guides.micronaut.io/latest/tag-spring-boot.html).

### CODE REVIEW

File: starter-core/src/test/groovy/io/micronaut/starter/springboot/SpringBootStarterSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import io.micronaut.starter.build.dependencies.Scope
+import io.micronaut.starter.fixture.CommandOutputFixture
+import io.micronaut.starter.options.BuildTool
+import io.micronaut.starter.options.Language
+import io.micronaut.starter.options.Options
+import io.micronaut.starter.options.TestFramework
+
+class SpringBootStarterSpec extends ApplicationContextSpec implements CommandOutputFixture {
+
+
```
Comment (timyates) [inline starter-core/src/test/groovy/io/micronaut/starter/springboot/SpringBootStarterSpec.groovy:17]:
```suggestion
    void "spring-boot is not included if it's not added as a framework"(Language language, BuildTool buildTool) {
        when:
        String template = new BuildBuilder(beanContext, buildTool)
                .language(language)
                .features([SpringBootStarter.NAME])
                .render()
        BuildTestVerifier verifier = BuildTestUtil.verifier(buildTool, language, template)

        then:
        verifier.hasBuildPlugin("io.micronaut.application")
        verifier.hasBuildPlugin("com.github.johnrengelman.shadow")
        !verifier.hasBuildPlugin("org.springframework.boot")
        !verifier.hasBuildPlugin("io.spring.dependency-management")

        where:
        [language, buildTool] << [[Language.JAVA], BuildTool.valuesGradle()].combinations()
    }

```

Is it worth adding a test to check when framework is not set, that the features do not get used?

**Review by timyates — APPROVED** [review]

**Comment by timyates** [comment]
There's a problem with AWS and Maven

https://ge.micronaut.io/s/6rrayn3g5cebu/tests/overview?outcome=failed

**Review by timyates — CHANGES_REQUESTED** [review]
https://ge.micronaut.io/s/6rrayn3g5cebu/tests/overview?outcome=failed

**Review by timyates — APPROVED** [review]
Fix pushed https://github.com/micronaut-projects/micronaut-starter/pull/1840/commits/253f78d5932d2700f6d5023e7cf2a1093a14c5e6

File: starter-core/src/test/groovy/io/micronaut/starter/springboot/SpringBootStarterSpec.groovy (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/test/groovy/io/micronaut/starter/springboot/SpringBootStarterSpec.groovy:17]: I don't think it is necessary. All these java classes are in the test classpath. We test that io.micronaut.application is not applied thus it is testing that if you use a framework different than the one selected the default features of that framework are not applied.

**Pull request merged**